### PR TITLE
welcome: Get rid of all if/else in the run() method and move to specific methods

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by OleksandrOrlov** [comment]
@okurz, I've already answered you in etherpad, but will duplicate it here (just to keep all in single place):
Not exactly. It wraps our code into a methods that actually add more readability, but do not solve the problem that we are forced to support all the possible scenarios, architectures and product versions in the same test module.

File: tests/installation/welcome.pm [hunk 1]
```diff
@@ -18,18 +18,49 @@ use testapi;
...
+        # Press Ok to confirm scc url
+        wait_screen_change { send_key 'alt-o' };
+        next;
+    }
+    if (match_has_tag 'inst-welcome-confirm-self-update-server') {
+        wait_screen_change { send_key $cmd{ok} };
+        next;
+    }
+    if (match_has_tag 'no-product-found-on-scc') {
+        record_soft_failure 'bsc#1056413';
```
Comment (asmorodskyi) [inline tests/installation/welcome.pm]: bug has verified / fixed status so need to drop this section

File: tests/installation/welcome.pm [hunk 2]
```diff
@@ -42,80 +73,66 @@ sub run {
...
-    }
+    handle_popups while ($iterations++ < scalar(@welcome_tags));
+}
 
-    # Process beta warning if expected
-    if ($expect_beta_warn) {
-        assert_screen 'inst-betawarning';
-        wait_screen_change { send_key 'ret' };
-    }
+sub process_beta_warning_if_expected {
```
Comment (asmorodskyi) [inline tests/installation/welcome.pm]: we have function which suppose to handle all popups, why we need a separate function to handle beta popup ? why not to do it all in one place ?

File: tests/installation/welcome.pm (hunk 1 above)
Comment (okurz) [reply tests/installation/welcome.pm]: sure, but not related to this PR, could you handle this?

File: tests/installation/welcome.pm (hunk 2 above)
Comment (okurz) [reply tests/installation/welcome.pm]: to make it clear from product perspective that we do something different here

File: tests/installation/welcome.pm (hunk 1 above)
Comment (asmorodskyi) [reply tests/installation/welcome.pm]: sure thing if rebasing and solve conflicts is more fun for you :) 
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6145

**Comment by rwx788** [comment]
Agreed with @OleksandrOrlov, it's adds readability, but doesn't simplify any logic.

**Comment by okurz** [comment]
@OleksandrOrlov Well, I am looking forward to a proposal how it could be done better

**Pull request closed without merging**